src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypeStyleDiagnosticAnalyzerBase.cs

```diff
@@ -91,6 +92,24 @@ private void HandleVariableDeclaration(SyntaxNodeAnalysisContext context)
                     shouldAnalyze = IsStylePreferred(semanticModel, optionSet, state, cancellationToken);
                 }
             }
+            else if (declarationStatement.IsKind(SyntaxKind.Argument))
+            {
+                shouldAnalyze = false;
+                declaredType = null;
+
+                if ((declarationStatement as ArgumentSyntax)?.Expression is DeclarationExpressionSyntax declaration)
```

Comment:
Instead of registering on Argument and then checking if the Argument.Expression is a DeclarationExpression, we should just registered to be called back on DeclarationExpressions.  Bonus is that we'll get far less callbacks if we do that.